### PR TITLE
Add Secret Manager Secret Data Source

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/9202

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_secret_manager_secret`
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/data_source_secret_manager_secret_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package google
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceSecretManagerSecret_basic(t *testing.T) {
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/data_source_secret_manager_secret_test.go:9]: This test failed with a `no "id" found in attributes` error

**Review by c2thorn — CHANGES_REQUESTED** [review]
Going over PR's I missed. This needs a rebase as well

File: mmv1/third_party/terraform/website/docs/d/secret_manager_secret.html.markdown [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+subcategory: "Secret Manager"
+layout: "google"
+page_title: "Google: google_secret_manager_secret"
+sidebar_current: "docs-google-datasource-secret-manager-secret"
+description: |-
+  Get information about a Secret Manager Secret
+---
+
+# google\_sql\_database\_instance
```
[the remark this answers is not in the document]
Comment (upodroid) [reply mmv1/third_party/terraform/website/docs/d/secret_manager_secret.html.markdown:10]: Fixed

File: mmv1/third_party/terraform/tests/data_source_secret_manager_secret_test.go (hunk 1 above)
Comment (upodroid) [reply mmv1/third_party/terraform/tests/data_source_secret_manager_secret_test.go:9]: Is it similar to this? https://github.com/GoogleCloudPlatform/magic-modules/pull/4515#issuecomment-783549514

**Comment by upodroid** [comment]
bump

File: mmv1/third_party/terraform/data_sources/data_source_secret_manager_secret.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	addOptionalFieldsToSchema(dsSchema, "project")
+
+	return &schema.Resource{
+		Read:   dataSourceSecretManagerSecretRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceSecretManagerSecretRead(d *schema.ResourceData, meta interface{}) error {
+
```
Comment (c2thorn) [inline mmv1/third_party/terraform/data_sources/data_source_secret_manager_secret.go:20]: Had some time to look into this, and realized that this function doesn't set an ID for the datasource. This should address the test failure from before.

File: mmv1/third_party/terraform/tests/data_source_secret_manager_secret_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccSqlDatabaseInstanceDestroyProducer(t),
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/data_source_secret_manager_secret_test.go:19]:
```suggestion
		CheckDestroy: testAccCheckSecretManagerSecretDestroyProducer(t),
```

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/data_sources/data_source_secret_manager_secret.go (hunk 3 above)
Comment (upodroid) [reply mmv1/third_party/terraform/data_sources/data_source_secret_manager_secret.go]: This is fixed now.

**Comment by upodroid** [comment]
Is it me or is the CI a bit wobbly this month?

**Comment by c2thorn** [comment]
@upodroid we made a lot of changes at the beginning of the month in attempts to improve things but also broke a few things along the way... Most of them should be fixed, but for this PR I'm having issues with the VCR build that I haven't seen in anywhere else.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**